src/lib/zf/tcp_out.c

```diff
@@ -956,7 +957,7 @@ int tcp_output(struct zf_tcp* tcp)
       else if( tcp_hdr->fin ) {
         flags |= ZF_PKT_REPORT_TCP_FIN;
       }
-      if( tcp_seg_seq(seg) != pcb->snd_nxt ) {
+      if( is_retransmit ) {
```

Comment:
This change is a bit non-obvious. Why did this have to be moved like this?

Reply:
This specific check to know if we're retransmitting didn't actually need to be moved, but the important part is `seg->in_flight`. I felt this was a better way to represent the code rather than storing another bool, e.g., `seg_was_in_flight`. Importantly, we must capture the state of `seg->in_flight` __before__ the call to `tcp_output_segment` in this function, which will result in `seg->in_flight` being set. I will add a comment to the definition of `is_retransmit` to explain exactly what's going on so will defer explanation to that comment rather than this PR to make sure the code is understandable in absence of this PR.